src/Extensions/Compatability/Validation/Code.cs

```diff
@@ -0,0 +1,489 @@
+		/// <typeparam name="TException">Exception to throw</typeparam>
+		/// <param name="state">Object's state</param>
+		/// <param name="errorMessage">Error message</param>
+		/// <param name="tagId">Tag Id to log, leave null if no logging is needed</param>
+		public static void Expects<TException>(bool state, string errorMessage, EventId? tagId)
+			where TException : Exception
+		{
+			if (!Validate(state, errorMessage, tagId))
+			{
+				ReportError<TException>(errorMessage);
+			}
+		}
+
+
+		/// <summary>
+		/// Checks that the enumerable argument is not null and doesn't contain any nulls
+		/// </summary>
+		/// <remarks>Be careful to not pass enumerables that can be enumerated only once</remarks>
+		/// <typeparam name="T">Type of the enumerable</typeparam>
+		/// <param name="argumentValue">The argument value.</param>
+		/// <param name="argumentName">Name of the argument.</param>
+		/// <param name="tagId">Tag Id to log, leave null if no logging is needed</param>
+		/// <exception cref="ArgumentException">Thrown if any argument  <paramref name="argumentValue"/> element is null.</exception>
+		/// <exception cref="ArgumentNullException">Thrown if the supplied argument <paramref name="argumentValue"/> is null.</exception>
+		public static IEnumerable<T> ExpectsAllNotNull<T>([ValidatedNotNull] IEnumerable<T>? argumentValue, string argumentName, EventId? tagId)
+			where T : class
+		{
+			argumentValue = ExpectsArgumentNotNull(argumentValue, argumentName, tagId);
+
+			if (!ValidateAllNotNull(argumentValue, argumentName, tagId))
+			{
+				ReportArgumentError(argumentName, AllErrorMessage);
+			}
+
+			return argumentValue;
+		}
+
+
+		/// <summary>
+		/// Checks the collection and throws an exception if it is null, contains no values, or contains any null values
+		/// </summary>
+		/// <typeparam name="T">Type of the collection</typeparam>
+		/// <param name="argumentValue">The argument value.</param>
+		/// <param name="argumentName">Name of the argument.</param>
+		/// <param name="tagId">Tag Id to log, leave null if no logging is needed.</param>
+		/// <exception cref="ArgumentException">Thrown if <paramref name="argumentValue"/> is empty or if any element is null.</exception>
+		/// <exception cref="ArgumentNullException">Thrown if the supplied argument <paramref name="argumentValue"/> is null.</exception>
+		public static IEnumerable<T> ExpectsNotEmptyAndAllNotNull<T>([ValidatedNotNull] ICollection<T>? argumentValue, string argumentName, EventId? tagId) where T : class
+		{
+			argumentValue = ExpectsArgumentNotNull(argumentValue, argumentName, tagId);
+
+			if (!ValidateNotEmptyAndAllNotNull(argumentValue, argumentName, tagId))
+			{
+				ReportArgumentError(argumentName, HasAnyErrorMessage);
+			}
+
+			return argumentValue;
+		}
+
+
+		/// <summary>
+		/// Checks the argument value and throws an exception if it is null or contains no values.
+		/// </summary>
+		/// <param name="argumentValue">The argument value.</param>
```

Comment:
>argumentValue [](start = 19, length = 13)

Some of these comment blocks are missing the `<typeparam>` comment

Reply:
It's old code that I would like to deprecate but If it will stay here for some time I'll update comments and method signatures